tests/lib/lte_lc_api/src/lte_lc_api_test.c

```diff
@@ -0,0 +1,1061 @@
+	ret = lte_lc_modem_events_enable();
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT%%MDMEV=1", -NRF_ENOMEM);
+	ret = lte_lc_modem_events_enable();
+	TEST_ASSERT_EQUAL(-EFAULT, ret);
+}
+
+void test_lte_lc_modem_events_disable(void)
+{
+	int ret;
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT%%MDMEV=0", EXIT_SUCCESS);
+	ret = lte_lc_modem_events_disable();
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT%%MDMEV=0", -NRF_ENOMEM);
+	ret = lte_lc_modem_events_disable();
+	TEST_ASSERT_EQUAL(-EFAULT, ret);
+}
+
+void test_lte_lc_periodic_search_request(void)
+{
+	int ret;
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT%%PERIODICSEARCHCONF=3", EXIT_SUCCESS);
+	ret = lte_lc_periodic_search_request();
+	TEST_ASSERT_EQUAL(EXIT_SUCCESS, ret);
+
+	__wrap_nrf_modem_at_printf_ExpectAndReturn("AT%%PERIODICSEARCHCONF=3", -NRF_ENOMEM);
+	ret = lte_lc_periodic_search_request();
+	TEST_ASSERT_EQUAL(-EFAULT, ret);
+}
+
+void test_lte_lc_conn_eval_params_get_null(void)
+{
+	int ret;
+
+	ret = lte_lc_conn_eval_params_get(NULL);
+	TEST_ASSERT_EQUAL(-EINVAL, ret);
+}
+
+void test_lte_lc_conn_eval_params_get_func_fail(void)
+{
+	int ret;
+	struct lte_lc_conn_eval_params params;
+
+	__wrap_custom_nrf_modem_at_scanf_ExpectAndReturn("AT+CFUN?", "+CFUN: %hu", -NRF_ENOMEM);
+	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_uint16(LTE_LC_FUNC_MODE_NORMAL);
+
+	ret = lte_lc_conn_eval_params_get(&params);
+	TEST_ASSERT_EQUAL(-EFAULT, ret);
+}
+
+void test_lte_lc_conn_eval_params_wrong_func_mode(void)
+{
+	int ret;
+	struct lte_lc_conn_eval_params params;
+
+	__wrap_custom_nrf_modem_at_scanf_ExpectAndReturn("AT+CFUN?", "+CFUN: %hu", 1);
+	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_uint16(LTE_LC_FUNC_MODE_OFFLINE);
+
+	ret = lte_lc_conn_eval_params_get(&params);
+	TEST_ASSERT_EQUAL(-EOPNOTSUPP, ret);
+}
+
+void test_lte_lc_conn_eval_params_at_fail(void)
+{
+	int ret;
+	struct lte_lc_conn_eval_params params;
+
+	__wrap_custom_nrf_modem_at_scanf_ExpectAndReturn("AT+CFUN?", "+CFUN: %hu", 1);
+	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_uint16(LTE_LC_FUNC_MODE_NORMAL);
+
+	__wrap_custom_nrf_modem_at_scanf_ExpectAndReturn(
+		"AT%CONEVAL",
+		"%%CONEVAL: %d,%hu,%hu,%hd,%hd,%hd,\"%x\",\"%6[^\"]\",%hd,%d,%hd,%hu,%hu,%hd,%hd,%hd,%hd",
+		-NRF_ENOMEM);
+	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_uint16(0);
+
+	ret = lte_lc_conn_eval_params_get(&params);
+	TEST_ASSERT_EQUAL(-EFAULT, ret);
+}
+
+void test_lte_lc_conn_eval_params_coneval_fail(void)
+{
+	int ret;
+	struct lte_lc_conn_eval_params params;
+
+	__wrap_custom_nrf_modem_at_scanf_ExpectAndReturn("AT+CFUN?", "+CFUN: %hu", 1);
+	__wrap_custom_nrf_modem_at_scanf_ReturnVarg_uint16(LTE_LC_FUNC_MODE_NORMAL);
+
+	__wrap_custom_nrf_modem_at_scanf_ExpectAndReturn(
+		"AT%CONEVAL",
+		"%%CONEVAL: %d,%hu,%hu,%hd,%hd,%hd,\"%x\",\"%6[^\"]\",%hd,%d,%hd,%hu,%hu,%hd,%hd,%hd,%hd",
+		17);
```

Comment:
When `%CONEVAL` fails, there's no cell information. I think `nrf_modem_at_scanf()` should return 1 in this case as the response is only `%CONEVAL: 1` and only one argument was matched?
```suggestion
		1);
```